apps/api/src/user/routes/user-api-keys/user-api-keys.router.ts

```diff
@@ -0,0 +1,208 @@
+import { createRoute } from "@hono/zod-openapi";
+import { container } from "tsyringe";
+import { z } from "zod";
+
+import { OpenApiHonoHandler } from "@src/core/services/open-api-hono-handler/open-api-hono-handler";
+import { UserApiKeyController } from "@src/user/controllers/user-api-keys/user-api-keys.controller";
+import {
+  CreateUserApiKeyRequestSchema,
+  FindUserApiKeyParamsSchema,
+  UpdateUserApiKeyRequestSchema,
+  UserApiKeyResponseSchema
+} from "@src/user/http-schemas/user-api-key.schema";
+
+const listRoute = createRoute({
+  method: "get",
+  path: "/v1/users/{userId}/api-keys",
+  summary: "List all API keys",
+  tags: ["User API Keys"],
+  request: {
+    params: z.object({
+      userId: z.string().uuid()
+    })
+  },
+  responses: {
+    200: {
+      description: "Returns list of API keys",
+      content: {
+        "application/json": {
+          schema: z.object({
+            data: z.array(UserApiKeyResponseSchema)
+          })
+        }
+      }
+    }
+  }
+});
+
+const getRoute = createRoute({
+  method: "get",
+  path: "/v1/users/{userId}/api-keys/{id}",
+  summary: "Get API key by ID",
+  tags: ["User API Keys"],
+  request: {
+    params: FindUserApiKeyParamsSchema
+  },
+  responses: {
+    200: {
+      description: "Returns API key details",
+      content: {
+        "application/json": {
+          schema: z.object({
+            data: UserApiKeyResponseSchema
+          })
+        }
+      }
+    },
+    404: {
+      description: "API key not found",
+      content: {
+        "application/json": {
+          schema: z.object({
+            message: z.string()
+          })
+        }
+      }
+    }
+  }
+});
+
+const postRoute = createRoute({
+  method: "post",
+  path: "/v1/users/{userId}/api-keys",
+  summary: "Create new API key",
+  tags: ["User API Keys"],
+  request: {
+    params: z.object({
+      userId: z.string().uuid()
+    }),
+    body: {
+      content: {
+        "application/json": {
+          schema: z.object({
+            data: CreateUserApiKeyRequestSchema
+          })
+        }
+      }
+    }
+  },
+  responses: {
+    201: {
+      description: "API key created successfully",
+      content: {
+        "application/json": {
+          schema: z.object({
+            data: UserApiKeyResponseSchema
+          })
+        }
+      }
+    }
+  }
+});
+
+const patchRoute = createRoute({
+  method: "patch",
+  path: "/v1/users/{userId}/api-keys/{id}",
```

Comment:
following up on the prev comment this endpoint should be also simplified:


```suggestion
  path: "/v1/api-keys/{id}",
```